public/app/features/dashboard/api/pubDashApi.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseQueryFn, createApi, retry } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime/src';
+import { notifyApp } from 'app/core/actions';
+import { createSuccessNotification } from 'app/core/copy/appNotification';
+import { PublicDashboard } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils';
+import { DashboardModel } from 'app/features/dashboard/state';
+
+const backendSrvBaseQuery =
+  ({ baseUrl }: { baseUrl: string } = { baseUrl: '' }): BaseQueryFn<BackendSrvRequest> =>
+  async (requestOptions) => {
+    try {
+      const { data: responseData, ...meta } = await lastValueFrom(
+        getBackendSrv().fetch({ ...requestOptions, url: baseUrl + requestOptions.url })
+      );
+      return { data: responseData, meta };
+    } catch (error) {
+      return { error };
+    }
+  };
+
+export const pubDashApi = createApi({
```

Comment:
Naming is hard. Not sure if we want to mix PublicDashboard and PubDash naming in components. For consistency, I would keep PublicDashboard. Maybe for variables it is fine. 
For APIs methods maybe we could remove the reference to pubdash if it belongs to PublicDashboardApi?

Reply:
you're right about api name. 
On the other hand, I wouldn't remove pubdash methods names because when you use those hooks it's not intuitive where they come from